### PR TITLE
Improve `startingDirectory` functionality

### PR DESCRIPTION
**Pull request by fghzxm**
This commit adds the `startingDirectory` property to the default-created
`cmd` and `powershell` profiles, with the default value
`%HOMEDRIVE%%HOMEPATH%`.

Signed-off-by: Fred Miller <fghzxm@outlook.com>

### CODE REVIEW

**Comment by DHowett-MSFT** [comment]
Should we use %USERPROFILE% Instead? We’ve already got a constant for it :)
Thanks!

________________________________

**Comment by fghzxm** [comment]
@DHowett-MSFT I inspected the lnks to cmd.exe and powershell.exe in my Start Menu, and they all have `%HOMEDRIVE%%HOMEPATH%` as their starting directories.  I wanted to match that behavior as strictly as possible.

File: src/cascadia/TerminalApp/CascadiaSettings.cpp [hunk 1]
```diff
@@ -123,6 +122,7 @@ void CascadiaSettings::_CreateDefaultProfiles()
     Profile defaultProfile{};
     defaultProfile.SetFontFace(L"Consolas");
     defaultProfile.SetCommandline(L"cmd.exe");
+    defaultProfile.SetStartingDirectory(L"%HOMEDRIVE%%HOMEPATH%");
```
Comment (binarycrusader) [inline src/cascadia/TerminalApp/CascadiaSettings.cpp]: HOMEDRIVE and HOMEPATH may not be set in all scenarios.  If you look at powershell core it has to deal with this today.  So unless an empty result here is valid, this logic probably needs to be centralized.

Comment (binarycrusader) [reply src/cascadia/TerminalApp/CascadiaSettings.cpp]: See notes here:
https://stackoverflow.com/posts/36392591/revisions

**Comment by binarycrusader** [comment]
> @DHowett-MSFT I inspected the lnks to cmd.exe and powershell.exe in my Start Menu, and they all have `%HOMEDRIVE%%HOMEPATH%` as their starting directories. I wanted to match that behavior as strictly as possible.

Correct:
image

**Comment by DHowett-MSFT** [comment]
The notes on the SO post suggest that perhaps USERPROFILE is safer? userenv is way earlier than explorer, and I’d almost certainly trust it more :)

________________________________

**Comment by fghzxm** [comment]
BTW, should we use `%SystemRoot%\system32\cmd.exe` and `%SystemRoot%\system32\WindowsPowerShell\v1.0\powershell.exe` instead of simply `cmd.exe` and `powershell.exe`?  (I guess we'll want path qualification if we want to have seperate 64-bit and 32-bit PowerShell profiles.)

**Comment by DHowett-MSFT** [comment]
> BTW, should we use `%SystemRoot%\system32\cmd.exe` and `%SystemRoot%\system32\WindowsPowerShell\v1.0\powershell.exe` instead of simply `cmd.exe` and `powershell.exe`? (I guess we'll want path qualification if we want to have seperate 64-bit and 32-bit PowerShell profiles.)

Ah, almost certainly. I'm on the fence about `pwsh.exe` (for the powershell core installation), because I imagine that people put the right one on their `%PATH%`... what do you think?

**Comment by fghzxm** [comment]
> I'm on the fence about pwsh.exe (for the powershell core installation), because I imagine that people put the right one on their %PATH%... what do you think?

@DHowett-MSFT But PowerShell Core isn't shipped with Windows, so I guess we actually don't want a PowerShell Core profile to be created by default?  (That would definitely be very confusing to users who don't have pwsh installed, wouldn't it)

**Comment by fghzxm** [comment]
> I'm on the fence about pwsh.exe (for the powershell core installation), because I imagine that people put the right one on their %PATH%... what do you think?

@DHowett-MSFT  Wait, I don't think PowerShell Core is shipped with Windows.

**Comment by zadjii-msft** [comment]
@fghzxm we actually have some code in the terminal already that attempts to detect if powershell core is installed, and uses `pwsh` instead of `powershell` if it is :)

File: src/cascadia/TerminalApp/CascadiaSettings.cpp [hunk 2]
```diff
@@ -123,6 +122,7 @@ void CascadiaSettings::_CreateDefaultProfiles()
     Profile cmdProfile{};
     cmdProfile.SetFontFace(L"Consolas");
     cmdProfile.SetCommandline(L"cmd.exe");
+    cmdProfile.SetStartingDirectory(L"%USERPROFILE%");
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/CascadiaSettings.cpp]:
```suggestion
    cmdProfile.SetStartingDirectory(DEFAULT_STARTING_DIRECTORY);
```

File: src/cascadia/TerminalApp/CascadiaSettings.cpp [hunk 3]
```diff
@@ -143,6 +143,7 @@ void CascadiaSettings::_CreateDefaultProfiles()
     }
     powershellProfile.SetFontFace(L"Courier New");
     powershellProfile.SetCommandline(psCmdline);
+    powershellProfile.SetStartingDirectory(L"%USERPROFILE%");
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/CascadiaSettings.cpp]:
```suggestion
    powershellProfile.SetStartingDirectory(DEFAULT_STARTING_DIRECTORY);
```

**Pull request merged**